### DISCUSSION TITLE
Possibility of specifying Investment accounts

### DISCUSSION THREAD

**Discussion by sohamshanbhag** [ananthakumaran/paisa#75] [Ideas]
Presently, paisa lumps all not in Assets:Checking as Investments. A suggestion would be to instead specify an/multiple investment account. This would allow accounts like Assets:Savings and other accounts people would like to use. The present implementation requires Assets:Checking:Savings, and similarly for other accounts. The justification on the website is as follows:

> The distinction is necessary because Checking account is an asset, but not an investment. So you ideally you want to disregard them when you calculate your absolute returns etc.

But Assets:Savings and other accounts would be used for absolute return calculation, which should not be.

A possible way to do this would be to specify the investment account like Assets:Investments:{Debt,Equity,...}, and add a configuration in paisa.yaml which allows one to specify this.

Checking the source code, Assets:{Equity,Debt} are not hardcoded anywhere. Regarding hardcoding Assets:Checking, this would require changes in internal/server/{cash_flow,investment,gain,expense,budget}.go to start.

**Comment by ananthakumaran** [comment]
This is a valid point and I will definitely bring account level configuration. In fact, it's already there in the master to support account-level icon customization. Every account has some behaviors, there are already 5 broad categories (Income, Assets, Liabilities, Expenses and Equity). Within each category, there are different types of behaviors like (income, refund), (savings, checking, investment), (credit card, loan), (tax, charges, expenses) etc. For now, I have supported a few of these behaviors with hardcoded naming conventions as you mentioned. I will make these configurable, but it will take some time. First priority is to have good support for the top 5 broad categories, once that is sorted, will get into the subcategories.
